spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteDataFilesSparkAction.java

```diff
@@ -106,7 +106,7 @@ public class RewriteDataFilesSparkAction
   private boolean caseSensitive;
 
   RewriteDataFilesSparkAction(SparkSession spark, Table table) {
-    super(spark.cloneSession());
+    super(((org.apache.spark.sql.classic.SparkSession) spark).cloneSession());
```

Comment:
I'm not sure this is safe to do, again I think we may have an issue with Spark Connect right? Don't we need to protect against these casts?

Reply:
the cast should be safe, in spark connect arch, the current iceberg spark connector is a side server plugin, it is not supposed to be invoked on the client side.